tests/integration_tests/performance/test_network_tcp_throughput.py

```diff
@@ -0,0 +1,367 @@
+    THROUGHPUT, DURATION, RETRANSMITS, CPU_UTILIZATION_HOST, \
+    CPU_UTILIZATION_GUEST, CONFIG, CPU_UTILIZATION_VMM, \
+    CPU_UTILIZATION_VCPUS_TOTAL, THROUGHPUT_TOTAL, RETRANSMITS_TOTAL, \
+    DURATION_TOTAL, IPERF3, BASE_PORT, DEBUG
+
+
+def measurements_tcp():
+    """Define the produced measurements for TCP workloads."""
+    return [types.MeasurementDef(THROUGHPUT, "Mbps"),
+            types.MeasurementDef(DURATION, "seconds"),
+            types.MeasurementDef(RETRANSMITS, "#"),
+            types.MeasurementDef(CPU_UTILIZATION_HOST, "percentage"),
+            types.MeasurementDef(CPU_UTILIZATION_GUEST, "percentage")]
+
+
+def stats_tcp(iperf3_id: str, env_id: str):
+    """Define statistics for TCP measurements."""
+    baseline_bw = CONFIG["baseline_bw"][env_id][iperf3_id]
+    delta_throughput = baseline_bw["delta"] * baseline_bw["target"] / 100
+    baseline_cpu_util = CONFIG["baseline_cpu_utilization"][env_id]
+    baseline_cpu_host = baseline_cpu_util[CPU_UTILIZATION_VMM][iperf3_id]
+    baseline_host_target = baseline_cpu_host["target"]
+    baseline_host_delta = \
+        baseline_cpu_host["delta"] * baseline_host_target / 100
+    baseline_cpu_guest = \
+        baseline_cpu_util[CPU_UTILIZATION_VCPUS_TOTAL][iperf3_id]
+    baseline_guest_target = baseline_cpu_guest["target"]
+    baseline_guest_delta = \
+        baseline_guest_target * baseline_guest_target / 100
+
+    return [types.StatisticDef(THROUGHPUT_TOTAL, THROUGHPUT, function.Sum,
+                               criteria.EqualWith(baseline_bw["target"],
+                                                  delta_throughput)),
+            types.StatisticDef(RETRANSMITS_TOTAL, RETRANSMITS, function.Sum),
+            types.StatisticDef(DURATION_TOTAL, DURATION, function.Avg),
+            types.StatisticDef(CPU_UTILIZATION_VMM,
+                               CPU_UTILIZATION_HOST,
+                               function.Identity,
+                               criteria.EqualWith(baseline_host_target,
+                                                  baseline_host_delta)),
+            types.StatisticDef(CPU_UTILIZATION_VCPUS_TOTAL,
+                               CPU_UTILIZATION_GUEST,
+                               function.Identity,
+                               criteria.EqualWith(baseline_guest_target,
+                                                  baseline_guest_delta))
+            ]
+
+
+def produce_iperf_output(basevm,
+                         guest_cmd_builder,
+                         current_avail_cpu,
+                         vcpus_count,
+                         runtime,
+                         omit,
+                         load_factor,
+                         modes):
+    """Produce iperf raw output from server-client connection."""
+    # Start the servers.
+    for determinant in range(load_factor*vcpus_count):
+        assigned_cpu = CpuMap(current_avail_cpu % vcpus_count)
```

Comment:
For a `load_factor` of one this will probably assign the same CPU core to a FC thread and to iperf3 server processes as well. I try to avoid this by having this method aware of other processes/threads already pinned to CPU cores, which are not available anymore.

Reply:
Right, sorry.